pennylane/grouping/pauli_group.py

```diff
@@ -0,0 +1,273 @@
+            used in the Pauli word as keys, and unique integer labels as their values.
+            If no wire map is provided, wires will be labeled by integers between 0 and ``n_qubits``.
+
+    Returns:
+        (qml.Operation): The next Pauli word in the group.
+
+    **Example**
+
+    The ``pauli_group_generator`` can be used to loop over the Pauli group as follows:
+
+    .. code-block:: python
+
+        from pennylane.grouping.pauli_group import pauli_group_generator
+
+        n_qubits = 3
+
+        for p in pauli_group_generator(n_qubits):
+            print(p)
+
+    Alternatively, using a custom wire map,
+
+    .. code-block:: python
+
+        from pennylane.wires import Wires
+        from pennylane.grouping.pauli_group import pauli_group_generator
+
+        n_qubits = 3
+        wire_map = {'a' : 0, 'b' : 1, 'c' : 2}
+
+        for p in pauli_group_generator(n_qubits, wire_map=wire_map):
+            print(p)
+
+    """
+    if not isinstance(n_qubits, int):
+        raise TypeError("Must specify an integer number of qubits construct the Pauli group.")
+
+    if n_qubits <= 0:
+        raise ValueError("Number of qubits must be at least 1 to construct Pauli group.")
+
+    element_idx = 0
+
+    if not wire_map:
+        wire_map = {wire_idx: wire_idx for wire_idx in range(n_qubits)}
+
+    while element_idx < 4 ** n_qubits:
+        binary_string = format(element_idx, f"#0{2*n_qubits+2}b")[2:]
+        binary_vector = [float(b) for b in binary_string]
+        yield binary_to_pauli(binary_vector, wire_map=wire_map)
+        element_idx += 1
+
+
+def pauli_group(n_qubits, wire_map=None):
+    """Constructs the n-qubit Pauli group.
+
+    This function constructs and returns the complete :math:`n`-qubit Pauli
+    group (without global phases) on the desired set of wires. This function
+    differs from ``pauli_group_generator`` in that returns the group in full
+    rather than one element at a time.
+
+    Args:
+        n_qubits (int): The number of qubits for which to create the group.
+        wire_map (dict[Union[str, int], int]): dictionary containing all wire labels used in the Pauli
+            word as keys, and unique integer labels as their values. If no wire map is
+            provided, wires will be labeled by integers between 0 and ``n_qubits``.
+
+    Returns:
+        (list[qml.Operation]): The full n-qubit Pauli group.
+
+    **Example**
+
+    Construction of the Pauli group can be done as follows:
+
+    .. code-block:: python
+
+        from pennylane.grouping.pauli_group import pauli_group
+
+        n_qubits = 3
+        pg = pauli_group(n_qubits)
+    """
+    if not isinstance(n_qubits, int):
+        raise TypeError("Must specify an integer number of qubits construct the Pauli group.")
+
+    if n_qubits <= 0:
+        raise ValueError("Number of qubits must be at least 1 to construct Pauli group.")
+
+    return list(pauli_group_generator(n_qubits, wire_map=wire_map))
```

Comment:
The fact that this function is simply calling `pauli_group_generator` but with `list` makes me think again we should simply delete this function, and make `pauli_group_generator` the default.